gpt_index/indices/response/response_builder.py

```diff
@@ -375,41 +344,248 @@ async def _aget_response_tree_summarize(
     ) -> RESPONSE_TEXT_TYPE:
-        """Get response."""
-        if mode == ResponseMode.DEFAULT:
-            return self._get_response_default(query_str, prev_response)
-        elif mode == ResponseMode.COMPACT:
-            return self._get_response_compact(query_str, prev_response)
-        elif mode == ResponseMode.TREE_SUMMARIZE:
-            return self._get_response_tree_summarize(
-                query_str, prev_response, **response_kwargs
+        text_qa_template = self._text_qa_template.partial_format(query_str=query_str)
+        node_text = self._service_context.prompt_helper.get_text_from_nodes(
+            [Node(text=text) for text in text_qa_template], 
+            prompt=text_qa_template
+        )
+
+        if not self._streaming:
+            (
+                response,
+                formatted_prompt,
+            ) = await self._service_context.llm_predictor.apredict(
+                text_qa_template,
+                context_str=node_text,
+            )
+            self._log_prompt_and_response(
+                formatted_prompt, response
+            )
+        else :
+            response, formatted_prompt = self._service_context.llm_predictor.stream(
+                text_qa_template,
+                context_str=node_text,
+            )
+            self._log_prompt_and_response(
+                formatted_prompt, response
             )
+        if isinstance(response, str):
+            response = response or "Empty Response"
         else:
-            raise ValueError(f"Invalid mode: {mode}")
+            response = cast(Generator, response)
+
+        return response
+
+
+    def get_response(
+        self, 
+        query_str: str, 
+        text_chunks: Sequence[str], 
+        prev_response: Optional[str] = None,
+    ) -> RESPONSE_TEXT_TYPE:
+        text_qa_template = self._text_qa_template.partial_format(query_str=query_str)
+        node_text = self._service_context.prompt_helper.get_text_from_nodes(
+            [Node(text=text) for text in text_chunks], 
+            prompt=text_qa_template
+        )
+
+        if not self._streaming:
+            (
+                response,
+                formatted_prompt,
+            ) = self._service_context.llm_predictor.predict(
+                text_qa_template,
+                context_str=node_text,
+            )
+            self._log_prompt_and_response(
+                formatted_prompt, response
+            )
+        else :
+            response, formatted_prompt = self._service_context.llm_predictor.stream(
+                text_qa_template,
+                context_str=node_text,
+            )
+            self._log_prompt_and_response(
+                formatted_prompt, response
+            )
+        if isinstance(response, str):
+            response = response or "Empty Response"
+        else:
+            response = cast(Generator, response)
+
+        return response
+
+
+class Generation(BaseResponseBuilder):
```

Comment:
is this purely for streaming use cases?

Reply:
This came from refactoring `GPTEmptyIndexQuery`. Where there's no retrieval logic, only generation based on query string.

Reply:
Oooooh. I see.